s3/pkg/datastore/yig/storage/gc.go

```diff
@@ -27,32 +28,47 @@ import (
 
 const (
 	GC_OBJECT_LIMIT_NUM = 10000
+	// max interval time of gc in seconds.
+	GC_MAX_INTERVAL_TIME = 3600
 )
 
 type GcMgr struct {
 	// context to cancel the operations.
 	ctx        context.Context
 	cancelFunc context.CancelFunc
 	yig        *YigStorage
-	loopTime   int64
-	wg         sync.WaitGroup
+	// the interval time of performing gc.
+	loopTime int64
+	wg       sync.WaitGroup
 }
 
 func (gm *GcMgr) Start() {
 	// query the current available cpus
 	threadNum := runtime.GOMAXPROCS(0)
 	gm.wg.Add(1)
 	go func() {
+		// loopCount is the number of loops for performing gc.
```

Comment:
this looks tricky, please share the requirement and design doc please, thanks

Reply:
this was already discussed in slack before, also, shufang had sent the design pictures in that conversation. pls check the conversation in the slack, thanks.